Integrate Makefile into automation

glower opened the pull request:
* Integrated Makefile into automation processes
* Updated documentation about Makefile usage
* Added documentation about LXD build
* Added missing Makefile targets for frontend tests and for code vendoring

glower commented:
I added two new github actions to run frontend tests and backend tests. Currently we use TravisCI for this tasks. I would suggest to deactivate TravisCI integration. In my experience github actions are much faster and more reliable.

glower commented on run-checks:
```diff
@@ -166,13 +164,10 @@ if [ ! -z "$UNIT" ]; then
     mkdir .coverage
     echo "mode: $COVERMODE" > .coverage/coverage.out
 
-    echo Building
```
we are building in a separate action, so don't need to do this here

doismellburning commented on README.md:
```diff
@@ -29,20 +31,46 @@ in order to changes take effect. That could require a browser restart.
 ## Install from Source
-If you have a Go development environment set up, Go get it, we recommend at least Go v1.13 or higher.
+
+If you have a Go development environment set up, we recommend at least Go v1.13 or higher. This project is build 
```
s/build/built/

doismellburning commented on README.md:
````diff
@@ -29,20 +31,46 @@ in order to changes take effect. That could require a browser restart.
+├── serial-vault-admin
+```
+
+To run Serial Vault in admin/UI mode you will need `static/` folder. Make sure to setup 
````
s/need/need a/

doismellburning commented on README.md:
````diff
@@ -29,20 +31,46 @@ in order to changes take effect. That could require a browser restart.
+```
+
+To run Serial Vault in admin/UI mode you will need `static/` folder. Make sure to setup 
+correct path to this folder in the configuration file with `docRoot` variable.
````
s/with/with the/

doismellburning commented on README.md:
```diff
@@ -112,10 +142,13 @@ To remove a dependency
 ### Install Go
-Follow the instructions to [install Go](https://golang.org/doc/install).
+
+Follow the instructions to [install Go](https://golang.org/doc/install). The current version of the service runns with `Go 1.13`.
```
s/runns/runs/

doismellburning commented on README.md:
```diff
@@ -29,20 +31,46 @@ in order to changes take effect. That could require a browser restart.
-If you have a Go development environment set up, Go get it, we recommend at least Go v1.13 or higher.
+
+If you have a Go development environment set up, we recommend at least Go v1.13 or higher. This project is build 
+and tested on Ubuntu 20.04.1 LTS` (Focal Fossa) with `go 1.13`. If you with to build or run the service from source we
```
s/with/wish/

doismellburning commented on README.md:
```diff
@@ -29,20 +31,46 @@ in order to changes take effect. That could require a browser restart.
+
+If you have a Go development environment set up, we recommend at least Go v1.13 or higher. This project is build 
+and tested on Ubuntu 20.04.1 LTS` (Focal Fossa) with `go 1.13`. If you with to build or run the service from source we
+recommend using LXD contaner for this purpose. To get started with LXD, please follow this 
```
s/container/container/

doismellburning reviewed (changes requested):
There is quite a lot going on here that makes it more challenging to meaningfully review than I'd like; could you try to keep it to commit-per-logical-change please? Trying to follow what you're trying to do is hard when e.g. the README diff has everything from formatting changes to adding LXD instructions to moving to Make usage etc.

Splitting it up would not only make it easier for me to review, it will help when looking back at history, trying to understand intent, etc. - I'd appreciate it if you would please.

glower commented:
> Splitting it up would not only make it easier for me to review, it will help when looking back at history, trying to understand intent, etc. - I'd appreciate it if you would please.

Sure, make sense. I will split the automation changes and README changes

glower commented on run-checks:
```diff
@@ -13,11 +13,9 @@ set -eu
 if which goctest >/dev/null; then
     goctest="goctest"
 else
-    goctest="go test"
```
'-mod=vendor' tells go to use vendor code from the `vendor` folder

glower commented on .github/workflows/nodejs.yml:
```diff
@@ -14,14 +14,7 @@ jobs:
         node-version: 12.x
     - name: Build frontend
       run: |
-        cd webapp-admin
```
this is moved to the Makefile

glower commented on .travis.yml:
```diff
@@ -1,50 +1,26 @@
-sudo: required
```
this is all not needed, new config file is clean

glower commented on .travis.yml:
```diff
@@ -1,50 +1,26 @@
       script:
-        - npm run test:ci
-      after_success:  
-        curl -s https://codecov.io/bash | bash /dev/stdin -f junit.xml
```
codecov never worked for the frontend tests, removed

glower commented:
* New github actions are removed and will be introduced in the new PR
* Unnecessary changes in the README are removed
* Added missing Makefile targets for frontend tests and for code vendoring

Documentation about LXD container could me also moved to a separate PR but I think it make sense here too

Roadmaster commented on .travis.yml:
```diff
@@ -1,50 +1,26 @@
-  - true
-
-script: sh -v ./run-checks $TEST_SUITE
+dist: focal
```
Note that it'd be best to test on the same distro where the service is deployed. That said, it's OK to leave this as focal and we should instead change the build/deploy process to use focal :) (tl;dr no change needed here).

glower replied:
Yes, that makes sense. I will change the build/test process to use focal in the next PR

Roadmaster reviewed (approved):
LGTM, a tiny comment but should not block the merge.

I agree with Kristian's request to split this into meaningful commits but if the resulting diff is the same, shouldn't need to request another review.
